hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableDescriptor.java

```diff
@@ -345,4 +345,11 @@ default boolean matchReplicationScope(boolean enabled) {
     }
     return !enabled;
   }
+
+  /**
+   * Check if user scan snapshot enable flag of the table is true. If flag is false then
+   * SnapshotScannerHDFSAclController won't handle HDFS acls for users with table read permission
+   * @return true if user scan snapshot is enabled for this table
+   */
+  boolean isUserScanSnapshotEnabled();
```

Comment:
The TableDescriptor is @InterfaceAudience.Public  interfaces, right ?  and the snapshot acl is a optional coproccesor ...
I think we can not add an extra specific method in the PUBLIC interface,  once someone use this method, then devs can not remove or change this easily, we MUST deprecated it in a old major release, then remove it in another new major release.   It is troublesome.